coreclr compilation failure on Alpine Linux (Musl based), "atoll is missing exception specification 'throw()'"

<!--This is just a template - feel free to delete any and all of it and replace as appropriate.-->
Continuation of https://github.com/dotnet/source-build/issues/1882.

### Description

coreclr fails to compile on an Alpine Linux edge system due to `error: 'atoll' is missing exception specification 'throw()'`.

To reproduce:

1. Get an Alpine Linux edge system, this can be in Docker if need be
2. Clone this repo
3. Run `build.sh`

### Configuration

This is on Alpine Linux edge, which is basically a rolling release version of Alpine Linux.
This system uses the Musl libc rather than glibc, which is probably the reason this is failing.

Right now I'm trying to build this for x86_64, but I personally doubt it's architecture related.

### Regression?

This is the first time I'm trying it, so I don't know.

### Other information

```
In file included from /home/bart/Documents/Git/runtime/src/coreclr/src/pal/src/misc/tracepointprovider.cpp:18:
  In file included from /home/bart/Documents/Git/runtime/src/coreclr/src/pal/src/include/pal/palinternal.h:584:
  In file included from /usr/bin/../lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../include/c++/10.2.1/stdlib.h:30:
  /usr/include/stdlib.h:23:11: error: 'atoll' is missing exception specification 'throw()'
  long long atoll (const char *);
            ^
  /home/bart/Documents/Git/runtime/src/coreclr/src/pal/inc/pal.h:3892:33: note: previous declaration is here
  PALIMPORT long long int __cdecl atoll(const char *)
                                  ^
  In file included from /home/bart/Documents/Git/runtime/src/coreclr/src/pal/src/exception/remote-unwind.cpp:44:
  In file included from /home/bart/Documents/Git/runtime/src/coreclr/src/pal/src/include/pal/palinternal.h:584:
  In file included from /usr/bin/../lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../include/c++/10.2.1/stdlib.h:30:
  /usr/include/stdlib.h:23:11: error: 'atoll' is missing exception specification 'throw()'
  long long atoll (const char *);
            ^
  /home/bart/Documents/Git/runtime/src/coreclr/src/pal/inc/pal.h:3892:33: note: previous declaration is here
  PALIMPORT long long int __cdecl atoll(const char *)
                                  ^
```

--- a/src/coreclr/src/pal/src/include/pal/palinternal.h
+++ b/src/coreclr/src/pal/src/include/pal/palinternal.h
@@ -170,6 +170,7 @@ function_name() to call the system's implementation
 #define memset DUMMY_memset
 #define memmove DUMMY_memmove
 #define memchr DUMMY_memchr
+#define atoll DUMMY_atoll
 #define strlen DUMMY_strlen
 #define stricmp DUMMY_stricmp
 #define strstr DUMMY_strstr
@@ -357,6 +358,7 @@ function_name() to call the system's implementation
 #undef memset
 #undef memmove
 #undef memchr
+#undef atoll
 #undef strlen
 #undef strnlen
 #undef wcsnlen

